packages/flutter_tools/lib/src/ios/devices.dart

```diff
@@ -348,6 +348,64 @@ Your Xcode version may be too old for your iOS version.
   }
 }
 
+/// Decodes an encoded syslog string to a UTF-8 representation.
+///
+/// Apple's syslog logs are encoded in 7-bit form. Input bytes are encoded as follows:
+/// 1. 0x00 to 0x19: non-printing range. Some ignored, some encoded as <...>.
+/// 2. 0x20 to 0x7f: as-is, with the exception of 0x5c (backslash).
+/// 3. 0x5c (backslash): octal representation \134.
+/// 4. 0x80 to 0x9f: \M^x (using control-character notation for range 0x00 to 0x40).
+/// 5. 0xa0: octal representation \240.
+/// 6. 0xa1 to 0xf7: \M-x (where x is the input byte stripped of its high-order bit).
+/// 7. 0xf8 to 0xff: unused in 4-byte UTF-8.
+String decodeSyslog(String line) {
+  // UTF-8 values for \, M, -, ^.
+  const int kBackslash = 0x5c;
+  const int kM = 0x4d;
+  const int kDash = 0x2d;
+  const int kCaret = 0x5e;
+
+  // Mask for the UTF-8 digit range.
+  const int kNum = 0x30;
+
+  // Returns true when [byte] is within the UTF-8 7-bit digit range: 0x30-0x39.
```

Comment:
nit: I believe our style guide calls for \`\` around non-member parameters instead of \[\] ?

Reply:
Done.